### PR TITLE
[stable/openldap][WIP] Add support for passing TLS keys from secret

### LINKED ISSUE #11719: [stable/openldap] TLS config options
**Is this a request for help?**:

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
FEATURE REQUEST

Please expose options to mount a Kubernetes secret representing custom TLS certificates.  Currently the only option appears to be forking osixia/docker-openldap , adding the certificates into the  image and setting appropriate environment variables.  If an option to mount a TLS secret into the openLDAP container were present it would feel a lot more flexible and would allow scenarios like using cert-manger, etc.

**Version of Helm and Kubernetes**:

**Which chart**:
stable/openldap

**What happened**:

**What you expected to happen**:

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by insertjokehere**
**After further testing I've run into issues using LetsEncrypt certificates with this - please consider this a WIP**

#### What this PR does / why we need it:
Adds an easy way to secure OpenLDAP connections using TLS certificates. In particular, this makes it possible to use certificates generated by cert-manager.

#### Which issue this PR fixes
 - fixes #11719

#### Special notes for your reviewer:
The change to the service to not publish the LDAPS port unless `tls.enabled` might not be appropriate, happy to back that out if people want me to

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by askm3** [comment]
I think this is very useful for organizations with custom/self signed certificates.
I suggest to customize this "tls" option to provide the ability to add a custom CA as from [docker-openldap](https://github.com/osixia/docker-openldap#use-your-own-certificate) specifications.
`--env LDAP_TLS_CRT_FILENAME=my-ldap.crt`
`--env LDAP_TLS_KEY_FILENAME=my-ldap.key`
`--env LDAP_TLS_CA_CRT_FILENAME=the-ca.crt`
Maybe we can use a custom secret instead of a tls secret?

File: stable/openldap/templates/deployment.yaml [hunk 1]
```diff
@@ -43,6 +43,16 @@ spec:
           mountPath: /ldifworkingdir
       {{- end }}
       containers:
+        {{- if .Values.tls.enabled }}
+        - name: {{ .Chart.Name }}-tls
+          image: busybox
+          command: ['sh', '-c', 'cp /tls/* /certs; while true; do sleep 10000; done']
```
Comment (enis) [inline stable/openldap/templates/deployment.yaml:49]: This should be an init container.

**Pull request closed without merging**